src/cfnlint/core.py

```diff
@@ -104,10 +104,11 @@ def create_parser():
 
     # Alllow the template to be passes as an optional or a positional argument
     standard.add_argument(
-        'template', nargs='?', help='The CloudFormation template to be linted')
+        'template', nargs='*', help='The CloudFormation template to be linted')
     standard.add_argument(
-        '-t', '--template', metavar='TEMPLATE', dest='template_alt', help='The CloudFormation template to be linted')
-
+        '-t', '--template', metavar='TEMPLATE', dest='template_alt',
```

Comment:
`nargs='+'` and `type=space_separated_arg` seems redundant. The args will be split already when passed in and this might break paths with spaces? (I'm not sure though)

Reply:
yeah don't break paths with spaces

Reply:
They were coming in as an array per character when I removed the `type=space_separated_arg`

And this results in only one `-t` working which doesn't make sense to me either.
```
standard.add_argument(
        '-t', '--template', metavar='TEMPLATE', dest='template_alt',
        help='The CloudFormation template to be linted', nargs='+')
```

This didn't make a lot of sense to me either but I'm probably missing something.

Reply:
Here is an example of allowing multiple arguments:

```py
import argparse
parser = argparse.ArgumentParser()

parser.add_argument('templates', nargs='*')
parser.add_argument('-t', dest='template_alt', nargs='+')

args = parser.parse_args()

print(repr(args))
```

And it works:

```
$ python test.py -t "a b" c d "e f" -- "1 2" 3 4 "5 6"
Namespace(template_alt=['a b', 'c', 'd', 'e f'], templates=['1 2', '3', '4', '5 6'])
```

I think the issue is the unnecessary `default=[]` or `action='extend'`?

Reply:
We allowed the extend for multiple `-t`.  I did find an odd issue with the way the extend function worked that I fixed.  That should resolve the strings being split into multiple element issue.